service/src/main/java/org/ehrbase/service/CompositionServiceImp.java

```diff
@@ -164,6 +165,37 @@ private UUID createInternal(
             throw new InternalServerException(e);
         }
 
+        if (composition.getUid() instanceof ObjectVersionId objectVersionId) {
+
+            if (objectVersionId.isBranch()) {
+                throw new PreconditionFailedException(
+                        "Version branching is not supported: %s".formatted(objectVersionId.getValue()));
+            }
+            if (Integer.parseInt(objectVersionId.getVersionTreeId().getValue()) != 1) {
```

Comment:
Could be simpler: First check
`"1".equals(objectVersionId.getVersionTreeId().getValue()`
and then 
`objectVersionId.isBranch()` 
to see what is wrong